src/Aspire.Hosting/Dcp/KubernetesService.cs

```diff
@@ -54,6 +57,36 @@ internal sealed class KubernetesService(ILogger<KubernetesService> logger, IOpti
 
     public TimeSpan MaxRetryDuration { get; set; } = TimeSpan.FromSeconds(20);
 
+    public Task<T> GetAsync<T>(string name, string? namespaceParameter = null, CancellationToken cancellationToken = default)
+        where T : CustomResource
+    {
+        var resourceType = GetResourceFor<T>();
+
+        return ExecuteWithRetry(
```

Comment:
@martincostello @mitchdenny We should use polly right?

Reply:
Looking at the code it looks like there's Polly retries at one level, and then this is retrying manually around that. I would have thought that if you need to retry the retries, that yeah you could use Polly here two with a differently configured pipeline/strategy.

Reply:
I'd rather see us use Polly than hand rolling retry logic. As for retries within retries I'd try to avoid that and agree with what Martin says (but he is the Polly expert :P)

Reply:
The code inside `KubernetesService` (and its retry logic) predates the use of Polly inside Aspire hosting.

If we want to standardize on Polly, we can do this, but given that this change will probably need to be backported to P5, I suggest we open a separate issue for that and address in P6/GA timeframe.